pkg/validate/ownerreferences.go

```diff
@@ -35,42 +35,42 @@ func (o OwnerRef) ValidatePath(ctx context.Context) error {
 		return err
 	}
 
-	switch o.Build.GetAnnotations()[build.AnnotationBuildRunDeletion] {
-	case "true":
-		// if the buildRun does not have an ownerreference to the Build, lets add it.
-		for i := range buildRunList.Items {
-			buildRun := buildRunList.Items[i]
-
-			if index := o.validateBuildOwnerReference(buildRun.OwnerReferences); index == -1 {
-				if err := controllerutil.SetControllerReference(o.Build, &buildRun, o.Scheme); err != nil {
-					o.Build.Status.Reason = build.BuildReasonPtr(build.SetOwnerReferenceFailed)
-					o.Build.Status.Message = pointer.String(fmt.Sprintf("unexpected error when trying to set the ownerreference: %v", err))
-				}
-				if err = o.Client.Update(ctx, &buildRun); err != nil {
-					return err
+	if o.Build.Spec.Retention != nil && o.Build.Spec.Retention.AtBuildDeletion != nil {
+		switch *o.Build.Spec.Retention.AtBuildDeletion {
```

Comment:
I think that `switch` should become an `if` now. Eliminates also the default case.